galley/pkg/runtime/monitoring/monitoring.go

```diff
@@ -134,6 +176,38 @@ func newView(measure stats.Measure, keys []tag.Key, aggregation *view.Aggregatio
 	}
 }
 
+func getStateTypeConfigKeys() ([]tag.Key, error) {
+	var err error
+	if NamespaceTag, err = tag.NewKey(namespace); err != nil {
+		panic(err)
+	}
+	if NameTag, err = tag.NewKey(name); err != nil {
+		panic(err)
+	}
+
+	return []tag.Key{NamespaceTag, NameTag}, err
+}
+
+func registerNewStateTypeConfigView(collection resource.Collection) error {
+	collectionStr := collection.String()
+	collectionSplit := strings.Split(collectionStr, "\\")
+	len := len(collectionSplit)
+	if len == 0 {
+		return fmt.Errorf("Collection name %v is invalid", collection)
+	}
+	stateTypeConfigTotal[collection.String()] = stats.Int64(fmt.Sprintf("galley/runtime/state/%s_info", collectionSplit[len-1]),
```

Comment:
I think we should pre-calculate the stateTypeConfigTotal fields. 

Also, the collection can be accessed concurrently.

Reply:
Can we use the full collection name here? It will make things very consistent.

Reply:
Thanks..Fixed

Reply:
Instead of full collection names in the metric name, can we just make that a label(s) on the metric itself?

compare:

`galley_runtime_state_istio_networking_v1alpha3_destinationrules{namespace="foo"}`

to:

`galley_runtime_state_destinationrules{api="networking.istio.io", version="v1alpha3", namespace="foo"}`

This would allow easier aggregation across version changes, etc. moving forward.  Thoughts?

Reply:
We should look at the https://prometheus.io/docs/practices/naming/ doc and try to reflect that in our choices.

Reply:
kinds are not unique. We had "policy" conflicting at some point IIRC. 

How about something like:
"galley_istio_networking_destinationrules"

Would that work?

Reply:
Based on prometheus naming... should it be "galley_istio_networking_destinationrules_total"?

Reply:
@ozevren are you suggesting that Istio itself uses multiple CRDs with the same `kind` ?